File: 구현/상하좌우/sehoongim/udlr.py
def udlr(inputs):
    n = inputs.split("\n")[0]
    directions = inputs.split("\n")[1].split(" ")

    position = [1,1]
    for direction in directions:
        if direction == "R" and position[1] != n:
            position[1] += 1
        if direction == "L" and position[1] != 1:
            position[1] -= 1
        if direction == "U" and position[0] != 1:
            position[0] -= 1
        if direction == "D" and position[0] != n:
            position[0] += 1
    return position

print(udlr("5\nR R R U D D"))
print(udlr("5\nR D D U R"))

File: 구현/상하좌우/sehoongim/udlr.rb
def udlr input
  n = input.split("\n")[0]
  directions = input.split("\n")[1].split(" ")

  answer = [1,1]
  directions.each do |direction|
    if direction == "R" and answer[1] != n
      answer[1] += 1
    end
    if direction == "L" and answer[1] != 1
      answer[1] += 1
    end
    if direction == "U" and answer[0] != 1
      answer[0] += 1
    end
    if direction == "D" and answer[0] != n
      answer[0] += 1
    end
  end
  p answer
  return "#{answer[0]} #{answer[1]}"
end

udlr("5\nR R R U D D")